src/zjs_timers.c

```diff
@@ -131,7 +140,7 @@ static jerry_value_t add_timer_helper(const jerry_value_t function_obj,
     zjs_timer_t* handle = add_timer(interval, callback, this, repeat, argv, argc - 2);
     if (handle->callback_id == -1)
         return zjs_error("native_set_interval_handler: timer alloc failed");
-    jerry_set_object_native_handle(timer_obj, (uintptr_t)handle, zjs_timer_free_cb);
```

Comment:
Setting this to NULL doesn't mean that JrS isn't GC'ing the object, it just means you're not informed about it.

But anyway you're right, I forgot about that - the timer should not just get GC'd like that. However, I figured out the issue I think - see my next comment.